queue_job/jobrunner/runner.py

```diff
@@ -479,10 +481,11 @@ def wait_notification(self):
         # probably a bug
         _logger.debug("select() timeout: %.2f sec", timeout)
         if timeout > 0:
-            conns, _, _ = select.select(conns, [], [], timeout)
             if conns and not self._stop:
-                for conn in conns:
-                    conn.poll()
+                with select() as sel:
+                    for conn in conns:
+                        sel.register(conn, selectors.EVENT_READ)
+                    sel.select(timeout)
```

Comment:
Don't we need to keep the psycopg2 `conn.poll()` call ?

Reply:
I think we do. It could go at line 440, just before we check conn.notifies.

Reply:
Good catch!

Thank you!

Fixed in the last commit